Source/DafnyCore/Compilers/SinglePassCompiler.cs

```diff
@@ -1872,9 +1881,9 @@ protected class NullClassWriter : IClassWriter {
 
       foreach (MemberDecl memberx in c.Members) {
         var member = (memberx as Function)?.ByMethodDecl ?? memberx;
-        if (!member.IsStatic) {
-          thisContext = c;
-        }
+        // if (!member.IsStatic) {
+        thisContext = c;
+        // }
```

Comment:
?

Reply:
Ah part of a different attempt to fix the hackery around coercing the result from a generic method, will revert.